dev-docs/RFCs/v6.x/gpu-aggregation-enhancements.md

```diff
@@ -0,0 +1,112 @@
+
+  * minMaxBuffer (Buffer, optional) : Contains data for one pixel, with R channel contains min value of all aggregated grid cells and A channel contains max value of all aggregated grid cells. When `combineMinMax` is `false` this value will be `null`.
+
+  Note: `minBuffer`, `maxBuffer` and `minMaxBuffer` are usually consumed by setting them as uniform buffer object or read by the CPU.
+
+
+### Implementation
+
+#### weights aggregation
+
+Aggregator internally maintains one texture for each weights object and performs one aggregation loop for each weights object. One aggregation loop (involves setting up Framebuffer object and one draw). Weight values are written into R, G and B channels and a value 1.0 into A channel for count. `blendEquation` is set to [`op`, `gl.SUM`], where `op` is `gl.SUM`, `gl.MIN` or `gl.MAX` based on user provided `operation` parameter.
+
+If `needMinMax` is set another aggregation loop is performed on above results, more details discussed below.
+
+Once aggregation is performed this data is asynchronously read into corresponding buffer before proceeding to the next weights object.
+
+
+#### MEAN operation
+
+`MIN`, `MAX` AND `SUM` operations can be directly calculated using corresponding `blendEquation`. For calculating `MEAN`, we first need to calculate sums and counts. Then divide each grid cell's sum with corresponding count. We can perform aggregation using `SUM`, and then we can run simple division operation on GPU using `Transform` class like below :
+
+```
+ ...
+
+ const {sumBuffer} = aggregationResults;
+
+ const transform = new Transform(gl2, {
+   sourceBuffers: {
+     inValue: sumBuffer
+   },
+   vs: `\
+#version 300 es
+in vec4 inValue;
+out vec4 outValue;
+
+void main()
+{
+  // count is in 'w' channel.
+  outValue.xyz = inValue.xyz/inValue.w;
+},
+   feedbackMap: {
+     inValue: 'outValue'
+   },
+   varyings: ['outValue'],
+   elementCount
+ });
+
+ const meanBuffer = transform.getBuffer('outValue');
+
+```
+
+Applications can also avoid this step by using `SUM` as operation and then calculating mean as above in applications shader. But this option is provide as convenience.
+
+
+### needMinMax and combineMinMax
+
+When `needMinMax` set to true, data is aggregated into single pixel to calculate min and max values of weights. Two separate rendering operations are performed, one with setting `blendEquation = gl.MIN` and the other with `blendEquation = gl.MAX`. Resulting data is stored in two separate buffers `minBuffer` and `maxBuffer`. If the `size` was set to 3, each channel in these buffer corresponds to min and max values of these weights. Alpha channel is always used for count.
+
+When `combineMinMax` is set to true, we avoid second render operation by setting `blendEquation = [gl.MIN, gl.MAX]`, but caveat here is it only supported when size = 1.
```

Comment:
This is a WebGL limitation? Any links that could be added to the doc?

Reply:
Updated text with details on why is this limitation.